File: day02/part2.m
% Advent of Code 2022
% Day 2
% Part 2

% A Rock
% B Paper
% C Scissors

% X Rock
% Y Paper
% Z Scissors

% 0 Loss
% 3 Draw
% 6 Win
% % % % % % % %
clc
clear all
% % % % % % % %

pairs = fscanf(fopen('input.txt','r'),'%s',[2 Inf]);
numPairs = size(pairs,2);
scoreOutcome = zeros(1,numPairs);
scorePlay = zeros(1,numPairs);

for i = 1:numPairs
    pair = pairs(:,i);
    opponent = pair(1);
    personal = pair(2);
    outcome = '';

    switch opponent
        case 'A'
            if strcmp(personal,'X')
                outcome = 'DRAW';
                scorePlay(i) = 1;
            elseif strcmp(personal,'Y')
                outcome = 'WIN';
                scorePlay(i) = 2;
            else
                outcome = 'LOSS';
                scorePlay(i) = 3;
            end

        case 'B'
            if strcmp(personal,'X')
                outcome = 'LOSS';
                scorePlay(i) = 1;
            elseif strcmp(personal,'Y')
                outcome = 'DRAW';
                scorePlay(i) = 2;
            else
                outcome = 'WIN';
                scorePlay(i) = 3;
            end

        case 'C'
            if strcmp(personal,'X')
                outcome = 'WIN';
                scorePlay(i) = 1;
            elseif strcmp(personal,'Y')
                outcome = 'LOSS';
                scorePlay(i) = 2;
            else
                outcome = 'DRAW';
                scorePlay(i) = 3;
            end
    end

switch outcome
    case 'WIN'
        scoreOutcome(i) = 6;
    case 'DRAW'
        scoreOutcome(i) = 3;
    case 'LOSS'
        scoreOutcome(i) = 0;
end
end

totalScore = sum(scoreOutcome) + sum(scorePlay);
fprintf('The total score is %d!\n\n',totalScore)

File: day02/part2.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>
#include <numeric>
using namespace std;

int main() {
    std::string line;
    std::fstream file("input.txt");

    char tempPlay;
    char tempOpp;

    int roundScore = 0;
    int idxOpp = 0;
    int idxMe = 2;
    int tempScore = 0;

    vector<char> opponent;
    vector<char> myPlay;
    vector<int> scores;

    while (std::getline(file,line)) {
        tempOpp = line[idxOpp];
        tempPlay = line[idxMe];
        opponent.push_back (tempOpp);
        myPlay.push_back(tempPlay);

        if (tempOpp == 'A' && tempPlay == 'X') {
            // rock x scissors
            tempScore = 3 + 0;
            scores.push_back(tempScore);
        } else if (tempOpp == 'A' && tempPlay == 'Y') {
            // rock x rock
            tempScore = 1 + 3;
            scores.push_back(tempScore);
        } else if (tempOpp == 'A' && tempPlay == 'Z') {
            // rock x paper
            tempScore = 2 + 6;
            scores.push_back(tempScore);
        } else if (tempOpp == 'B' && tempPlay == 'X') {
            // paper x rock
            tempScore = 1 + 0;
            scores.push_back(tempScore);
        } else if (tempOpp == 'B' && tempPlay == 'Y') {
            // paper x paper
            tempScore = 2 + 3;
            scores.push_back(tempScore);
        } else if (tempOpp == 'B' && tempPlay == 'Z') {
            // paper x scissors
            tempScore = 3 + 6;
            scores.push_back(tempScore);
        } else if (tempOpp == 'C' && tempPlay == 'X') {
            // scissors x paper
            tempScore = 2 + 0;
            scores.push_back(tempScore);
        } else if (tempOpp == 'C' && tempPlay == 'Y') {
            // scissors x scissors
            tempScore = 3 + 3;
            scores.push_back(tempScore);
        } else if (tempOpp == 'C' && tempPlay == 'Z') {
            // scissors x rock
            tempScore = 1 + 6;
            scores.push_back(tempScore);
        }

    }

    int totalScore = std::accumulate(scores.begin(), scores.end(), 0);

    printf("The Total Score is: %i\n\n", totalScore);
}